wordpress-mobile/release-toolkit
# Update Fastlane to 2.148.1

While working on the Simplenote screenshots, I had made a dedicated lane to upload them to the App Store. It worked fine for at the time of 4.18, but then when I tried yesterday for 4.19 it failed.

```
Error loading plugin 'fastlane-plugin-wpmreleasetoolkit':
  uninitialized constant Faraday::Error::ClientError

Did you mean?  Faraday::ClientError
```

As far as I could trace back with Google searches, without looking at the changelogs of the gems, it looks like there was some incompatibility between `octokit` and `faraday` after certain updates. There are more details [in this issue](https://github.com/github/pages-gem/issues/665), sort of.

I updated `fastlane` to its latest version here, and then used the branch that was giving me trouble to verify the fix.

## To test

1. Download a set of of pre-generated Simplenote iOS screenshots [here](https://github.com/Automattic/simplenote-ios/suites/731298482/artifacts/7342385). It's a `.zip` file, which you'll need to extract.
1. Checkout [this branch on Simplenote iOS](https://github.com/Automattic/simplenote-ios/pull/765)
1. Run `bundle install && bundle exec fastlane create_promo_screenshots source:/path/to/your/extracted/screenshots`

You should get an error:

```
[✔] 🚀
[22:40:18]: Error loading plugin 'fastlane-plugin-wpmreleasetoolkit': uninitialized constant Faraday::Error::ClientError
Did you mean?  Faraday::ClientError
+-----------------------------------+---------+----------------------------------------+
|                                     Used plugins                                     |
+-----------------------------------+---------+----------------------------------------+
| Plugin                            | Version | Action                                 |
+-----------------------------------+---------+----------------------------------------+
| fastlane-plugin-wpmreleasetoolkit | 0.9.2   | No actions found                       |
```

Next, update the `Pluginfile` to point to this branch, `use-faraday-1.x`:

```diff
-gem 'fastlane-plugin-wpmreleasetoolkit', git: 'https://github.com/wordpress-mobile/release-toolkit', tag: '0.9.2'
+gem 'fastlane-plugin-wpmreleasetoolkit', git: 'https://github.com/wordpress-mobile/release-toolkit', branch: 'use-faraday-1.x'
```

Try again, `bundle install && bundle exec fastlane create_promo_screenshots source:/path/to/your/extracted/screenshots`, and verify it works.

Note, if you get

```
Git LFS not enabled – Unable to generate promo screenshots. Run `git lfs install && git lfs fetch && git lfs pull` to fix this.
```

that's actually good as it shows that some of the code that wasn't working before actually is 😄 . Run the `git lfs` commands and it should work properly.
